File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/ShippersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ShippersRow } from "./ShippersRow";

export interface ShippersColumns {
    MaShipper: Column<ShippersRow>;
    MaShipperCode: Column<ShippersRow>;
    TenCongTy: Column<ShippersRow>;
    TenShipper: Column<ShippersRow>;
    SoDienThoai: Column<ShippersRow>;
}

export class ShippersColumns extends ColumnsBase<ShippersRow> {
    static readonly columnsKey = 'SanPhamDB.Shippers';
    static readonly Fields = fieldsProxy<ShippersColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/Shippers/ShippersColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.Shippers")]
[BasedOnRow(typeof(ShippersRow), CheckNames = true)]
public class ShippersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaShipper { get; set; }
    [EditLink]
    public string MaShipperCode { get; set; }
    public string TenCongTy { get; set; }
    public string TenShipper { get; set; }
    public string SoDienThoai { get; set; }
}